Subpage for Custom Repository setup (#9887)
mendix/docs: content/en/docs/refguide/java-programming/managed-dependencies/custom-repositories.md

        // credentials {
        //    username = 'username'
        //    password = 'password'
        //}
    }

    // This is for the CycloneDx Gradle plugin
    maven {
       url = uri("http://localhost:8046/artifactory/gradle-remote/") // Make sure to use the correct url to your repository
       // If the repository requires authentication, then uncomment the following config
       // credentials {
       //    username = 'username'
       //    password = 'password'
       //}
    }
    ```
- Press Ok and Studio Pro will start syncing the dependencies for the project.
- If everything goes well, you'll be able to see your `vendorlib` directory populated with the required project dependencies and `vendorlib-sbom.json` file.
- Then you can run the app to verify if it works.

# How to set up a local Repository

## 1. Prerequisites

For scenarios where you have absolutely no access to the internet(air-gapped), you can create a folder on your local system which will contain all the project dependencies. Then you can configure Studio Pro to resolve dependencies from this folder.

### Downloading M2 repository

> We use [CycloneDx Gradle Plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin) to generate the `vendorlib-sbom.json` file. This is for Compliance and Security requirements. You will have to create a `m2` (Maven) style directory which will contain this plugin. You can download it from the [Mendix GitHub repo](broken-link)

Reviewer: Link to Mendix GitHub is missing

## 2. Configuring the repository

> Create a folder containing a flattened list of all the project dependencies. Make sure you store this on your local system where Studio Pro has read/write access to it.  

Reviewer: Can you elaborate on this? What does "flattened list" mean? How do I determine the project dependencies? Where do I download them? This could also use an example

Reviewer: Add a tip: let studio pro download this for you and copy the vendorlib folder

## 3. Configuring Studio Pro to use local repository

Once you have set up your local repository, you need to add a few configuration settings in Studio Pro. This will enable Studio Pro to resolve dependencies from the local repository.

- Go to `Edit -> Preferences -> Deployment -> Use custom repository`. Enable this option.
- Input the following settings into the text area field that will appear once your enable the above option
  ```gradle      
  flatDir {

>     dirs '../libs' // This is the path to the project level dependencies. It is recommended to use absolute paths. But relative path can also be used.

Reviewer: Change the example to an absolute path if that is recommended

       //}
    }
    ```

1. Click Ok and Studio Pro will start syncing the dependencies for the project.

    If everything goes well, you will see your `vendorlib` directory populated with the required project dependencies and a `vendorlib-sbom.json` file.

1. Run the app to verify if it works.

## How to Set up a Local Repository{#local-repo}

### Prerequisites

For scenarios where you have absolutely no access to the internet (air-gapped), you can create a folder on your local system which will contain all the project dependencies. Then you can configure Studio Pro to resolve dependencies from this folder.

#### Downloading the M2 repository

> You should use the [CycloneDx Gradle Plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin) to generate the `vendorlib-sbom.json` file. This is to meet compliance and security requirements. You will have to create an `m2` (Maven) style directory which will contain this plugin. You can download it from the [Mendix GitHub repo](https://github.com/mendix/m2Repo-with-cycloneDX)

Author: I still think this sentence should start with a "We". 

We are, unfortunately, using the plugin to generate the `sbom` file and customers are forced to use it. The message I tried to convey here is "why we are using it", which is for compliance reasons. 

You could also choose a different phrase to convey this message.

Reviewer 2: I think it should be "you" as it is something that the customer has to do. However, I agree that we need more information to explain why. How about:
```suggestion
You will have to create an `m2` (Maven) style directory which will contain the [CycloneDx Gradle Plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin). You can download the content for this directory from the [Mendix GitHub repo](https://github.com/mendix/m2Repo-with-cycloneDX). The CycloneDx Gradle plugin is used to meet compliance and security requirements when generating the `vendorlib-sbom.json` file.
```

Author: Yeah. I like this. Please go ahead with this change.
